File: 0274-h-index/0274-h-index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        # first the citations in reverse order
        citations.sort(reverse=True)
        
        # h-index
        h = 0

        if len(citations) == 1 and citations[0] == 0:
            return 0

        # the h index correlates to the sorted index
        for i, num in enumerate(citations):
            if num >= i+1:
                h = i+1
            if num < i:
                break;
        
        return h

            

File: 0274-h-index/0274-h-index.ts
function hIndex(citations: number[]): number {
    // if (citations.length === 1 && citations[0] == 0) return 0;
    // let h: number = 0;
    // citations.sort((a, b) => b - a);
    // for (let i:number = 0; i < citations.length; i ++) {
    //     if (citations[i] >= i + 1) {
    //         h = i + 1
    //     } else {
    //         break;
    //     }
    // };

    // return h;
    
    return citations.sort((a, b) => b - a).filter((val, i) => val > i).length;
};